scheduler/src/cook/mesos/task.clj

```diff
@@ -25,6 +25,30 @@
 (defonce custom-executor-name "cook_agent_executor")
 (defonce custom-executor-source "cook_scheduler")
 
+(defn- use-custom-executor?
+  "Returns true if the job should be scheduled to use the custom executor."
+  [job-ent]
+  (or (:job/custom-executor job-ent true)
+      (= :job.executor/custom (:job/executor job-ent))))
+
+(defn use-cook-executor?
+  "Returns true if the job should be scheduled to use the Cook executor.
+   Cook executor is used when the following conditions are true:
+   1. The job is not configured to use the custom executor (including backwards compatibility),
+   2. The Cook executor command has been configured
+   3. Either :job/executor is explicitly enabled
+      Or: a. Cook executor has not been explicitly disabled,
+          b. This is going to be the first instance of the job, and
+          c. Our random toss yields less than portion percent."
+  [job-ent executor-config]
+  (and (not (use-custom-executor? job-ent))
+       (:command executor-config)
+       (or (= :job.executor/cook (:job/executor job-ent))
+           (and (nil? (:job/executor job-ent))
+                (zero? (count (:job/instance job-ent)))
+                (when-let [portion (:portion executor-config)]
+                  (> (* portion 100) (rand-int 100)))))))
```

Comment:
It would be nice if this was deterministic based upon `job-ent`, e.g. `(<= (-> (:id job-ent) .getLeastSignificantBits (mod 100) (/ 100.0)) portion)`